runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -163,7 +165,7 @@ protected void onResponse(StreamingCommitResponse response) {
         continue;
       }
       PendingRequest done = pending.remove(requestId);
-      if (done == null) {
+      if (done == null && !isShutdown()) {
```

Comment:
this is going to get nullptr exception below, instead move the isShutdown check to whether or not to log inside this if

Reply:
done